lib/spack/spack/build_systems/octave.py

```diff
@@ -0,0 +1,70 @@
+##############################################################################
+# Copyright (c) 2013-2017, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/spack/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+import inspect
+import os
+
+from spack.directives import depends_on, extends
+from spack.package import PackageBase, run_after
+
+
+class OctavePackage(PackageBase):
+    """Specialized class for Octave packages. See
+    https://www.gnu.org/software/octave/doc/v4.2.0/Installing-and-Removing-Packages.html
+    for more information.
+
+    This class provides the following phases that can be overridden:
+
+    1. :py:meth:`~.OctavePackage.install`
+
+    """
+    # Default phases
+    phases = ['install']
+
+    # To be used in UI queries that require to know which
+    # build-system class we are using
+    build_system_class = 'OctavePackage'
+
+    extends('octave')
+    depends_on('octave', type=('build', 'run'))
+
+    def install(self, spec, prefix):
+        """Install the package from the archive file"""
+
+        # octave does not like those environment variables to be set:
+        os.environ.pop('CC', '')
+        os.environ.pop('CXX', '')
+        os.environ.pop('FC', '')
```

Comment:
I'm wondering if it would be better to separate this out into `setup_environment`.

Reply:
done.

Reply:
minor inconvenience is that we have warnings:
```
==> Warning: Suspicious requests to set or unset 'CC' found
==> Warning: 	    	env.set('CC', join_path(link_dir, compiler.link_paths['cc'])) at /Users/davydden/spack/lib/spack/spack/build_environment.py:146
==> Warning: 	--->	spack_env.unset('CC') at /Users/davydden/spack/lib/spack/spack/build_systems/octave.py:54
==> Warning: Suspicious requests to set or unset 'CXX' found
==> Warning: 	    	env.set('CXX', join_path(link_dir, compiler.link_paths['cxx'])) at /Users/davydden/spack/lib/spack/spack/build_environment.py:149
==> Warning: 	--->	spack_env.unset('CXX') at /Users/davydden/spack/lib/spack/spack/build_systems/octave.py:55
==> Warning: Suspicious requests to set or unset 'FC' found
==> Warning: 	    	env.set('FC', join_path(link_dir, compiler.link_paths['fc'])) at /Users/davydden/spack/lib/spack/spack/build_environment.py:155
==> Warning: 	--->	spack_env.unset('FC') at /Users/davydden/spack/lib/spack/spack/build_systems/octave.py:56
```
but I am ok with it.